src/application/clients/crux-wallet-client.ts

```diff
@@ -0,0 +1,257 @@
+    private cruxDomain?: CruxDomain;
+    private cruxUserRepository!: ICruxUserRepository;
+    private cruxAssetTranslator?: CruxAssetTranslator;
+    private keyManager?: IKeyManager;
+    private resolvedClientAssetMapping?: IResolvedClientAssetMap;
+    private cacheStorage?: StorageService;
+
+    constructor(options: ICruxWalletClientOptions) {
+        this.cacheStorage = options.cacheStorage || new InMemStorage();
+        this.cruxBlockstackInfrastructure = options.blockstackInfrastructure || CruxSpec.blockstack.infrastructure;
+        this.walletClientName = options.walletClientName;
+        if (options.privateKey) {
+            this.keyManager = typeof options.privateKey === "string" ? new BasicKeyManager(options.privateKey) : options.privateKey;
+        }
+        this.cruxDomainId = new CruxDomainId(this.walletClientName);
+        this.initPromise = this.init(options);
+    }
+
+    @throwCruxClientError
+    public getCruxIDState = async (): Promise<ICruxIDState> => {
+        await this.initPromise;
+        if (!this.keyManager) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.PrivateKeyRequired);
+        }
+        const cruxUser = await this.cruxUserRepository.getWithKey(this.keyManager, this.cruxDomainId);
+        if (!cruxUser) {
+            return {
+                cruxID: null,
+                status: {
+                    status: SubdomainRegistrationStatus.NONE,
+                    statusDetail: SubdomainRegistrationStatusDetail.NONE,
+                },
+            };
+        } else {
+            return {
+                cruxID: cruxUser.cruxID.toString(),
+                status : cruxUser.info.registrationStatus,
+            };
+        }
+    }
+
+    @throwCruxClientError
+    public resolveCurrencyAddressForCruxID = async (fullCruxID: string, walletCurrencySymbol: string): Promise<IAddress> => {
+        await this.initPromise;
+        const tag = "resolving_address";
+        const cruxUser = await this.getCruxUserByID(fullCruxID.toLowerCase(), tag);
+        if (!cruxUser) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.UserDoesNotExist);
+        }
+        const assetId = this.getCruxAssetTranslator().symbolToAssetId(walletCurrencySymbol);
+        if (!assetId) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.AssetIDNotAvailable);
+        }
+        const address =  cruxUser.getAddressFromAsset(assetId);
+        if (!address) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.AddressNotAvailable);
+        }
+        return address;
+    }
+
+    @throwCruxClientError
+    public getAddressMap = async (): Promise<IAddressMapping> => {
+        await this.initPromise;
+        if (!this.keyManager) {
+            throw ErrorHelper.getPackageError(null, PackageErrorCode.PrivateKeyRequired);
+        }
+        const cruxUser = await this.cruxUserRepository.getWithKey(this.keyManager, this.cruxDomainId);
```

Comment:
Is the behavior b/w old and new SDK same when cruxUser doesn't exist?

Reply:
same